### PR TITLE
exit with non-zero status code on warnings

### LINKED ISSUE #13: exit with non-zero code on warnings
First of all thank you for this great library!

Do you think it would make sense to exit with non-zero when there are warnings? For example, I'm using this in my CI, and would like to fail if dialyzer is not happy. Currently I have to do some bash gimmicks and analyze stderr output to make it happen. With the non-zero exit code then it would just work.

If you think this makes sense, I'd be happy to prepar a PR.

### PR DESCRIPTION
**Pull request by sasa1977**
resolves #13

### CODE REVIEW

File: lib/mix/tasks/dialyze.ex [hunk 1]
```diff
@@ -82,6 +82,7 @@ defmodule Mix.Tasks.Dialyze do
     else
       warnings ->
         print_warnings(warnings)
+        if (match?([_|_], warnings)), do: System.halt(1)
```
Comment (fishcakez) [inline lib/mix/tasks/dialyze.ex]: Can we raise a mix error instead, which includes the number of dialyer warnings?

Comment (sasa1977) [inline lib/mix/tasks/dialyze.ex]: Something like:

```elixir
 Mix.raise "Dialyzer reported #{length(warnings)} warnings"
```

?

Comment (fishcakez) [inline lib/mix/tasks/dialyze.ex]: Yes `Mix.raise/1`, the message can be whatever you like.

**Comment by sasa1977** [comment]
Pushed the commit

File: lib/mix/tasks/dialyze.ex [hunk 2]
```diff
@@ -82,6 +82,9 @@ defmodule Mix.Tasks.Dialyze do
     else
       warnings ->
         print_warnings(warnings)
+        if (match?([_|_], warnings)) do
+          Mix.raise "Dialyzer reported #{length(warnings)} warnings"
+        end
```
Comment (fishcakez) [inline lib/mix/tasks/dialyze.ex]: This returns nil from `run/1`, spec says `:ok`. I think it would be clearer to match on the else clause.

**Comment by sasa1977** [comment]
Here's another attempt

**Comment by fishcakez** [comment]
Thanks!

**Pull request merged**